pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -92,12 +96,17 @@ func (volumes *VolumesV1) createVolume(opts VolumeCreateOpts) (string, string, e
 
 	vol, err := volumes_v1.Create(volumes.blockstorage, create_opts).Extract()
 	if err != nil {
+		recordOpenstackOperationMetric("create_v1_volume", 0, err)
 		return "", "", err
 	}
+	timeToken := time.Since(startTime).Seconds()
+	recordOpenstackOperationMetric("create_v1_volume", timeToken, nil)
```

Comment:
I'm very reluctant to add anything complex on error paths because they're ~never tested - and there's rarely much point optimising the error case either...